mlflow/models/evaluation/base.py

```diff
@@ -31,6 +31,14 @@ def __init__(self, uri, content=None):
         self._uri = uri
         self._content = content
 
+    @staticmethod
+    @abstractmethod
+    def file_extension():
+        """File extension for this artifact type (e.g. "png", "csv", etc.)"""
+        raise NotImplementedError(
+            "File extension must be provided for implementations of EvaluationArtifact"
```

Comment:
Though it's generally a good idea to have file extension in file name, I'm not fully convinced all artifacts have to come with an extension. If I want to store some byte string in base64 encoded format, it could be literally anything, e.g. a protobuf, a PNG, a serialized JSON etc.

Reply:
I guess my original reasoning was that since every artifact corresponds to a file, it should have a file extension. For example, in the case of a byte string in base64 artifact, its file extension will be whatever file type it is being stored as. Like .proto, .png, .json. Does that make sense?